src/script/lua_api/l_env.cpp

```diff
@@ -751,7 +751,8 @@ int ModApiEnvMod::l_set_timeofday(lua_State *L)
 
 	// Do it
 	float timeofday_f = readParam<float>(L, 1);
-	sanity_check(timeofday_f >= 0.0 && timeofday_f <= 1.0);
+	luaL_argcheck(L, timeofday_f >= 0.0 && timeofday_f <= 1.0, 1,
```

Comment:
Although many areas of code do not do this yet, float literals should be stated with an 'f'. So:
0.0f
1.0f